tests/system/src/test/java/test/com/sun/marlin/DashedRectTest.java

```diff
@@ -133,84 +171,29 @@ public void start(Stage primaryStage) throws Exception {
             path2.getStrokeDashArray().setAll(DASH_LEN);
 
             Scene scene = new Scene(new Group(path1, path2));
-            stage.setScene(scene);
-            stage.setTitle("DashedRectTest");
-            stage.show();
 
-            launchLatch.countDown();
-        }
-    }
+            myApp.stage.setScene(scene);
 
-    @BeforeClass
-    public static void setupOnce() {
-        // Start the Application
-        new Thread(() -> Application.launch(MyApp.class, (String[]) null)).start();
-
-        try {
-            if (!launchLatch.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
-                throw new AssertionFailedError("Timeout waiting for Application to launch");
-            }
-
-        } catch (InterruptedException ex) {
-            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
-            err.initCause(ex);
-            throw err;
-        }
-
-        assertEquals(0, launchLatch.getCount());
-    }
-
-    @AfterClass
-    public static void teardownOnce() {
-        Platform.exit();
-    }
-
-    @Test(timeout = 10000)
-    public void TestDashedPath() throws InterruptedException, IOException {
-
-        final int size = MAX * 2;
-
-        final WritableImage img = new WritableImage(size, size);
-
-        done = false;
-        Platform.runLater(() -> {
             final SnapshotParameters sp = new SnapshotParameters();
             sp.setViewport(new Rectangle2D(0, 0, size, size));
 
-            myApp.stage.getScene().getRoot().snapshot(sp, img);
-
-            signalDone();
+            scene.getRoot().snapshot(sp, img);
```

Comment:
I just noticed that you aren't using the return value from snapshot. According to spec, you should not assume that it will always use the image that is passed in (it will use it if you can). This should be:

```
    img = scene.getRoot().snapshot(sp, img);
```

If you put the check into the runAndWait as well, then you don't need to make img final.

Reply:
fixed